cds/modules/deposit/api.py

```diff
@@ -174,6 +180,38 @@ def patch(self, *args, **kwargs):
         """Patch only drafts."""
         return super(CDSDeposit, self).patch(*args, **kwargs)
 
+    @property
+    def report_number(self):
+        try:
+            return self['report_number']['report_number']
+        except KeyError:
+            return None
+
+    @report_number.setter
+    def report_number(self, value):
+        self['report_number'] = dict(
+            report_number=value,
+            _report_number=None,
+        )
+
+    def generate_report_number(self, **kwargs):
+        """Generates a new report number."""
+        self.report_number = report_number_minter(self.id, self, **kwargs)
+
+    @property
+    def sequence_name(self):
+        """Get the name of the corresponding sequence for this class."""
+        return self.__class__.__name__.upper()
```

Comment:
What do you think about using a class attribute instead? we have cases with two submissions using the same report number template but different workflows

```python
class Video(CDSDeposit):

    report_number_template_name = 'video-1_0_0' 
    ...
```